src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateCopies = [];
+  let stateCopy = { ...state };
+
+  for (const action of actions) {
+    let nextState;
+
+    switch (action.type) {
+      case 'addProperties':
+        nextState = { ...stateCopy, ...action.extraData };
+        break;
+      case 'removeProperties':
+        nextState = { ...stateCopy };
+
+        for (const key of action.keysToRemove) {
+          if (nextState.hasOwnProperty(key)) {
+            delete nextState[key];
+          }
+        }
+        break;
+      case 'clear':
+        nextState = {};
+        break;
+    }
```

Comment:
The switch statement is missing a default case. Always include a default case in switch statements for handling unexpected values. This can help with debugging and ensuring that the function behaves correctly even if it encounters an unknown action type.